src/v/config/configuration.cc

```diff
@@ -480,6 +480,26 @@ configuration::configuration()
        .visibility = visibility::user},
       model::timestamp_type::create_time,
       {model::timestamp_type::create_time, model::timestamp_type::append_time})
+  , log_message_timestamp_alert_before_ms(
+      *this,
+      "log_message_timestamp_alert_before_ms",
+      "Threshold in milliseconds for alerting on messages with a timestamp "
+      "after the broker's time, meaning they are in the future relative to the "
+      "broker's clock.",
+      {.needs_restart = needs_restart::no,
+       .example = "315360000000",
+       .visibility = visibility::tunable},
+      std::chrono::milliseconds::max())
```

Comment:
the defaults should be 2hrs ahead/behind the broker's time. that's a LONGGGG time.

Reply:
The reason to have this is because timestamps in the past are a legitimate use case for event sourcing. In the original design a check for the past wasn't considered, it was added to mirror [KIP-937](https://cwiki.apache.org/confluence/display/KAFKA/KIP-937%3A+Improve+Message+Timestamp+Validation)  https://github.com/redpanda-data/planning/issues/86

I'll refactor to have an optional here